### PR TITLE
cds new style

### PR DESCRIPTION
**Pull request by troy2914**

### CODE REVIEW

File: lastminute.yml [hunk 1]
```diff
@@ -320,11 +320,11 @@
       reverseproxy_server: true
       tags: reverseproxy
 
-- name: Configure the cds
+- name: Configure the cds instances (new sytle)
```
Comment (vmcj) [inline lastminute.yml:323]:
```suggestion
- name: Configure the cds instances (new style)
```

File: lastminute.yml [hunk 2]
```diff
@@ -320,11 +320,11 @@
...
   hosts: cds
   roles:
-    - role: cds
-      tags: cds
+    - role: cds-next
+      tags: cds-next
```
Comment (vmcj) [inline lastminute.yml:327]: I don't think renaming for the role is needed?

Comment (troy2914) [reply lastminute.yml]: fixed in last commit (rebased)

File: lastminute.yml (hunk 1 above)
Comment (troy2914) [reply lastminute.yml]: fixed in last commit (rebased)

File: roles/cds/tasks/cds_instance.yml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    src: templates/cds.service.j2
+    dest: /etc/systemd/system/{{ cds_instance_name }}.service
+    mode: '0644'
+  register: unit_updated
+
+- name: "Update cds_needs_restart: after service for {{ cds_instance_name }}"
+  ansible.builtin.set_fact:
+    cds_needs_restart: "{{ cds_needs_restart | bool or unit_updated.changed }}"
+
+- name: Restart+enable cds instance
```
Comment (vmcj) [inline roles/cds/tasks/cds_instance.yml:102]: Why can't this be done in the handler anymore? The templates will do a notify on change, but what is the resource where the notify route does not work?

File: roles/cds/tasks/main.yml [hunk 4]
```diff
@@ -1,37 +1,92 @@
...
+  ansible.builtin.uri:
+    url: https://api.github.com/repos/icpctools/icpctools/releases?per_page=1
+    method: GET
+    return_content: true
+    status_code: 200
+    body_format: json
+  register: latest_cds_release_array
+  check_mode: false
+
+- name: Set CDS latest version
```
Comment (vmcj) [inline roles/cds/tasks/main.yml:29]: maybe we should put this whole github scraping section in a block, when the version is latest do the scraping if its set in a variable use that. We tend to use the latest at our contests but if/when that is broken pinning temporary in an easy way will make reuse easier. If you agree we can create an issue for this and I can do this (so it shouldn't be blocking for this PR)

File: roles/cds/tasks/cds_instance.yml (hunk 3 above)
Comment (ubergeek42) [reply roles/cds/tasks/cds_instance.yml:102]: There's no way to parameterize a handler, so notify doesn't work. We include this tasks file multiple times, once per instance (e.g. a normal cds instance and a p100 instance), but you can't make dynamic handlers (at least that I could find online). And you don't want to restart the regular CDS if you change the config for p100 so you can't just restart all of the instances.

File: roles/cds/tasks/main.yml (hunk 4 above)
Comment (ubergeek42) [reply roles/cds/tasks/main.yml:29]: Yeah. That seems reasonable to me.

File: roles/cds/files/authorized_keys2
Comment (ubergeek42) [inline roles/cds/files/authorized_keys2:1]: This file doesn't appear to be referenced, we should just delete it.

**Review by ubergeek42 — APPROVED** [review]

File: roles/cds/files/authorized_keys2
Comment (troy2914) [reply roles/cds/files/authorized_keys2:1]: Hmm, i think it was suppose to be installed to the new user running the cds.  eg for the rundeck and backup to scp/ssh to that user?

Comment (troy2914) [reply roles/cds/files/authorized_keys2:1]: removed, should be done in the future by cds for icpc role, and a rundeck role.

**Pull request merged**